dispatchers.go

```diff
@@ -187,9 +187,9 @@ func dispatchCreateOperation(
 		pusher(ctx.events...)
 	}
 
-	if ctx.outputData != nil {
-		evt := elemental.NewEvent(elemental.EventCreate, ctx.outputData.(elemental.Identifiable))
-		pusher(evt)
+	switch o := ctx.outputData.(type) {
+	case elemental.Identifiable:
```

Comment:
What happens if I have a nil identifiable?  Can you also add a test case if not already there?

Reply:
Would still work fine due to the fact that `nil` would be a valid value that an interface can hold - https://golang.org/ref/spec#Switch_statements:

> Instead of a type, a case may use the predeclared identifier nil; that case is selected when the expression in the TypeSwitchGuard is a nil interface value. There may be at most one nil case.

See https://play.golang.org/p/d4Atwl3Lysw

```go
package main

import (
	"fmt"
	"encoding/json"
)

func main() {
	switcher("hello") // prints "I am a string!"
	switcher(12) // prints "I am an int!"
	switcher(nil) // prints "I am nil!" 
	switcher(json.RawMessage("amir is awesome!")) // prints "I am something else!"
}

func switcher(i interface{}) {
	switch i.(type) {
	case string:
		fmt.Println("I am a string!")
	case int:
		fmt.Println("I am an int!")
	case nil:
		fmt.Println("I am nil!")
	default:
		fmt.Println("I am something else!")
	}
}
```

Reply:
The point is we don't want to perform a push if nil

Reply:
That seems like something that the `eventPusherFunc` should be responsible for checking then?

Reply:
Actually nvm my last comment, if `Identifiable` is `nil`, then it would cause a panic in elemental here - https://github.com/aporeto-inc/elemental/blob/master/event.go#L54

So no event would get pushed.

Reply:
And correct me if I am wrong, but the only way to pass in a `nil` `Identifiable` would be if you have a pointer to a type that satisfies that interface and the value of the interface is `nil`. So in that situation, you are clearly doing something strange and it should 💥

 Right?

Reply:
If you _really_ wanted to avoid the panic, then you could do something like this within the case statement above:

```go
if (reflect.ValueOf(o).Kind() == reflect.Ptr && !reflect.ValueOf(o).IsNil()) {
    pusher(elemental.NewEvent(elemental.EventCreate, o))
}
```

Reply:
An example of how to get a `nil` `elemental.Identifiable`:

```go
			// notice how this is a type that satisfies the elemental.Identifiable interface
			var testIdentifiable *someType = nil
			var _ elemental.Identifiable = testIdentifiable
			ctx.SetOutputData(testIdentifiable)
```

Reply:
And this will **NOT** cause a panic 💥 because you will never be able to pass in a `nil` interface if you follow the call path from [here](https://github.com/aporeto-inc/bahamut/pull/65/files/96aea1e1d8575701d0fba40090851b6c71be47d3#diff-5801991d7bcf817b83f880119a6985e9R192):

- `elemental.NewEvent` which will be passed a non-nil interface value, which calls
  - `NewEventWithEncoding` which will call `Encode` with a non-nil interface value
    - `Encode` will run, which will not trigger [this condition](https://github.com/aporeto-inc/elemental/blob/master/encoding.go#L148) as the interface value passed in is not `nil`

Reply:
To summarize the last few comments: passing in a `nil` `Identifiable` will not cause a panic.

Reply:
Add a test about that then :)

Reply:
Will do 😉

Reply:
Done!